### PR TITLE
DDF-2876 Enhance LDAP tests to use SSL

### PR DESCRIPTION
**Pull request by ahoffer**
#### What does this PR do?
Enhances LDAP unit tests. Creates an LDAP server that listens on a TCP socket and a TCP+SSL socket. The SslLdapLoginModule (the class under test) then connects to the server with different configurations and tests the results.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
 @codymacdonald @mweser @emanns95 @AzGoalie 
#### Select at least one member from relevant component team(s) from below (at least one 
@brendan-hofmann
#### How should this be tested? (List steps with links to updated documentation)
Run unit tests on DDF and DIB. 
Do a full build for good measure.
#### Any background context you want to provide?
This ticket is part of a larger effort to remove the embedded LDAP server. That also means removing the itests that depend on the embedded LDAP server. The ticket pushes the testing down, out of the itests and into the unit tests. However, the code is still tested against a live LDAP server.
#### What are the relevant tickets?
[DDF-2876](https://codice.atlassian.net/browse/DDF-2876)
#### Checklist:
- [N/A] Documentation Updated
- [N/A] Change Log Updated
- [X] Update / Add Unit Tests
- [N/A] Update / Add Integration Tests

### CODE REVIEW

**Comment by ahoffer** [comment]
build now

File: platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapLoginConfig.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 import java.util.Properties;
 import java.util.UUID;
 
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
```
Comment (brendan-hofmann) [inline platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapLoginConfig.java]: Why the switch to lang3 here and below?

File: platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java [hunk 2]
```diff
@@ -166,11 +164,13 @@ protected boolean doLogin() throws LoginException {
...
         // valid password (because if authentication = none, the password could be any
         // value - it is ignored).
-        if ("none".equals(authentication) && (tmpPassword != null)) {
+        // Username is not checked in this conditional because a null username immediately exits
+        // this method.
+        if ("none".equalsIgnoreCase(getBindMethod()) && (tmpPassword != null)) {
             LOGGER.debug(
                     "Changing from authentication = none to simple since user or password was specified.");
             // default to simple so that the provided user/password will get checked
-            authentication = "simple";
```
Comment (ahoffer) [inline platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java]: Unit testing caught an error in the code. The authentication variable was being set, but not used to decide what auth method should be used. Changed to setting the binding method property.

File: platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapLoginConfig.java (hunk 1 above)
Comment (ahoffer) [inline platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapLoginConfig.java]: More attention to security in lang3

**Comment by mweser** [comment]
Full swarm build successful.

**Comment by ahoffer** [comment]
build now

File: platform/security/sts/security-sts-ldaplogin/pom.xml [hunk 3]
```diff
@@ -82,17 +82,17 @@
                                         <limit>
                                             <counter>INSTRUCTION</counter>
                                             <value>COVEREDRATIO</value>
-                                            <minimum>0.72</minimum>
+                                            <minimum>0.7</minimum>
```
Comment (coyotesqrl) [inline platform/security/sts/security-sts-ldaplogin/pom.xml]: We usually specify these in precision to the hundredths. Any reason you chose tenths?

Comment (ahoffer) [inline platform/security/sts/security-sts-ldaplogin/pom.xml]: No special reason for tenths. I copied the values that jacoco printed to the command line and it happened to print in tenths.

**Pull request merged**